tensorboard/components/tf_backend/requestManager.ts

```diff
@@ -58,6 +58,30 @@ export class RequestNetworkError extends Error {
   }
 }
 
+/**
+ * Holds options that can be used to configure the HTTP request.
+ */
+export class RequestOptions {
+  /** The HTTP method-type to use. Currently only 'GET' and 'POST' are
+   * supported.
+   */
+  public methodType: string;
```

Comment:
Few things: 
1. This can be an enum with string value.
```ts
export enum MethodType {
  GET = 'GET',
  POST = 'POST,
}
```

2. It seems like there should be some preconditions. For instance, for POST request, body should be required (or does POST without body makes sense?) and there should be an explicit setting of Content-Type when body is present. Do you think these are good ideas? If so, would you mind implementing it?

3. Since this is one of the core library code, would you mind adding some test around the new public API?

Reply:
Thanks Stephan!
I've changed the string to an enum, and added some preconditions.
I've added tests and discovered issue #1990 along the way.
PTAL!